chart/validator-plugin-aws/templates/deployment.yaml

```diff
@@ -68,5 +71,15 @@ spec:
           | nindent 10 }}
       securityContext:
         runAsNonRoot: true
+      {{- if .Values.auth.serviceAccountName }}
+      serviceAccountName: {{ .Values.auth.serviceAccountName }}
+      {{- else }}
       serviceAccountName: {{ include "chart.fullname" . }}-controller-manager
-      terminationGracePeriodSeconds: 10
\ No newline at end of file
+      {{- end }}
+      terminationGracePeriodSeconds: 10
+      {{- if .Values.auth.secret }}
+      volumes:
+      - name: aws-auth
+        secret:
+          secretName: {{ .Values.auth.secret.secretName }}
```

Comment:
its possible that no `secretName` isnt defined when we enter this if block. is there any issue with that? ie should the if condition on line 80 be something like this?

```
{{- if and .Values.auth.secret .Values.auth.secret.secretName }}
```

Reply:
Yeah, theoretically someone could leave `secret.secretName` empty, but there isn't much we can do to protect against that.

Reply:
Actually, evidently I don't [know my template functions](https://helm.sh/docs/howto/charts_tips_and_tricks/#know-your-template-functions). Will fix this.